contracts/handler/CompatibilityFallbackHandler.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+
+import "./DefaultCallbackHandler.sol";
+import "../interfaces/ISignatureValidator.sol";
+import "../GnosisSafe.sol";
+
+
+/// @title Compatibility Fallback Handler - fallback handler to provider compatibility between pre 1.3.0 and 1.3.0+ Safe contracts
+/// @author Richard Meissner - <richard@gnosis.pm>
+contract CompatibilityFallbackHandler is DefaultCallbackHandler, ISignatureValidator {
+    address internal constant SENTINEL_MODULES = address(0x1);
+    bytes4 internal constant UPDATED_MAGIC_VALUE = 0x1626ba7e;
+
+    /**
+    * Implementation of ISignatureValidator (see `interfaces/ISignatureValidator.sol`)
+    * @dev Should return whether the signature provided is valid for the provided data.
+    * @param _data Arbitrary length data signed on the behalf of address(msg.sender)
+    * @param _signature Signature byte array associated with _data
+    * @return a bool upon valid or invalid signature with corresponding _data
+    */
+    function isValidSignature(bytes calldata _data, bytes calldata _signature)
```

Comment:
Why is this logic on the `CompatibilityFallbackHandler`?

Reply:
you mean it should be a separate callback handler? ... in general this is available on the current interface and will be removed, so to have compatibility we also support the old interface.